src/_pytest/assertion/util.py

```diff
@@ -131,6 +132,8 @@ def isiterable(obj: Any) -> bool:
 def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[str]]:
     """Return specialised explanations for some operators/operands"""
     verbose = config.getoption("verbose")
+    screen_width = config.get_terminal_reporter().screen_width
```

Comment:
Instead of adding `get_terminal_reporter()` to `Config` and a `screen_width` property, it will be better to use `config.get_terminal_writer().fullwidth`.